app/services/publisher_wallet_getter.rb

```diff
@@ -16,26 +16,24 @@ def perform
       request.url("/v1/owners/#{URI.escape(publisher.owner_identifier)}/wallet")
     end
 
-    channel_responses = {}
-    publisher.channels.verified.each do |channel|
-      identifier =  channel.details.channel_identifier
-      channel_responses[identifier] = connection.get do |request|
-        request.headers["Authorization"] = api_authorization_header
-        request.url("/v2/publishers/#{URI.escape(identifier)}/balance")
-      end
-    end
-
     wallet_hash = JSON.parse(wallet_response.body)
 
-    channel_hash = {}
-    channel_responses.each do |identifier, response|
-      channel_hash[identifier] = JSON.parse(response.body)
+    rates = wallet_hash["rates"]
+    if wallet_hash && wallet_hash["contributions"] && wallet_hash["contributions"]["currency"]
```

Comment:
I think you can do wallet_hash.dig("contributions", "currency") for readability

https://stackoverflow.com/questions/19115838/how-do-i-use-the-fetch-method-for-nested-hash

For max readability: `currency = wallet_hash.dig("contributions, "currency") || publisher.default_currency`